packages/components/src/combobox-control/index.js

```diff
@@ -81,6 +81,7 @@ function ComboboxControl( {
 	const [ inputHasFocus, setInputHasFocus ] = useState( false );
 	const [ inputValue, setInputValue ] = useState( '' );
 	const inputContainer = useRef();
+	const changingLabel = selectedSuggestion?.label || inputValue;
```

Comment:
Apart from that, the introduction of the `changingLabel` variable completely breaks the typing interaction in `ComboboxControl`, since the input won't show what's being typed by the user when in its expanded state and a value has already been selected

(the video below shows what just described)

https://user-images.githubusercontent.com/1083581/191969784-40b777b8-c1bd-4915-91cd-aa15230eb1d0.mp4